### PR TITLE
FEAT: Posthog Events across product

### PR DESCRIPTION
**Pull request by JayeshVP24**
## Screenshots (if appropriate)

### CODE REVIEW

**Review by a20hek — CHANGES_REQUESTED** [review]
`pnpm.lock` file should not have a diff
just pull the latest version from main

File: src/components/home/SideBar.tsx [hunk 1]
```diff
@@ -103,7 +105,12 @@ const GettingStarted = ({ userInfo }: GettingStartedProps) => {
               _hover={{
                 color: 'brand.purple',
               }}
-              onClick={() => !userInfo?.id && setTriggerLogin(true)}
+              onClick={() => {
+                if (!userInfo?.id) {
+                  posthog.capture('create account_getting started');
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/components/home/SideBar.tsx:111]: we are following the following naming convention here
(event name separated by spaces)(underscore)(location or feature name of event generated at separated by space)

**Review by a20hek — COMMENTED** [review]
have a question:
if we are capturing pageviews, then what's the need for capturing through useEffect for certain pages?

File: src/features/navbar/components/MobileNavbar.tsx [hunk 2]
```diff
@@ -252,11 +273,15 @@ export const MobileNavbar = ({ onLoginOpen, onSearchOpen }: Props) => {
         py={2}
         bg={'#F8FAFC'}
       >
-        <Flex gap={{ base: 2, sm: 8, md: 12 }}>
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/features/navbar/components/MobileNavbar.tsx:255]: might have accidentally merged old ones, switching back

**Comment by JayeshVP24** [comment]
> have a question: if we are capturing pageviews, then what's the need for capturing through useEffect for certain pages?

create funnels in posthog using PageView event is a not easy 
pratik asked for custom names only for the funnels he's making

and sometimes u want event name based on type of page like open_bounty and open_project are same page

File: src/features/listings/components/Submission/SubmissionActionButton.tsx [hunk 3]
```diff
@@ -92,13 +97,18 @@ export const SubmissionActionButton = ({
       if (!userInfo?.isTalentFilled) {
         warningOnOpen();
       } else {
+        if (buttonState === 'submit') {
+          console.log('submit');
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/features/listings/components/Submission/SubmissionActionButton.tsx]: done

File: src/features/listings/components/Submission/SubmissionActionButton.tsx [hunk 4]
```diff
@@ -92,13 +97,18 @@ export const SubmissionActionButton = ({
...
+          posthog.capture('start_submission');
+        } else if (buttonState === 'edit') {
+          console.log('edit');
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/features/listings/components/Submission/SubmissionActionButton.tsx]: done

File: src/features/navbar/components/MobileNavbar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/features/navbar/components/MobileNavbar.tsx:255]: done

File: src/features/navbar/components/MobileNavbar.tsx [hunk 5]
```diff
@@ -276,11 +301,15 @@ export const MobileNavbar = ({ onLoginOpen, onSearchOpen }: Props) => {
           borderColor={'brand.slate.400'}
           orientation="vertical"
         />
-        <Flex gap={{ base: 2, sm: 8, md: 12 }}>
+        <Flex className="ph-no-capture" gap={{ base: 2, sm: 12 }}>
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply src/features/navbar/components/MobileNavbar.tsx]: done

**Review by a20hek — APPROVED** [review]

**Pull request merged**